### PR TITLE
Check if there is package.json file in the directory

### PR DESCRIPTION
**Pull request by Tolsee**
Solves: [https://github.com/alex-saunders/glicky/issues/5]

Throw an error and show a message, if there is no `package.json` file in the directory

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -9,9 +9,19 @@ const getPort = require('get-port');
...
 ('use strict');
 
+// Log Utility
+function logWarning(...params) {
+  console.log(chalk.yellow.bold(...params));
+}
+
+function logError(...params) {
+  console.log(chalk.red.bold(...params));
+}
```
Comment (alex-saunders) [inline bin/index.js]: Thanks for these refactors, it makes things much cleaner ✨

File: bin/index.js [hunk 2]
```diff
@@ -105,6 +107,11 @@ const { open, port } = minimist(process.argv.slice(2), {
 });
 (async () => {
   try {
+    const { pkg } = await readPkgUp();
+    if (!pkg || !Object.keys(pkg).length) {
+      logError('This directory does not contain package.json file (via npm init).');
```
Comment (alex-saunders) [inline bin/index.js]: Could we possibly reword this message to explain the problem a bit more?
Maybe something like this:
```suggestion
      logError('At the moment, Glicky does not support projects that have not been initialised with a package.json file. Please run `npm init` before running Glicky in this directory.');
```

**Comment by alex-saunders** [comment]
@Tolsee I checked out this PR locally and it looks like it might not be working 100% as intended. I think read-pkg-up will actually traverse parent directories until it finds a package.json file, so I ran glicky from an empty directory and it managed to find a package.json somewhere up the directory tree:

Screenshot 2019-04-18 at 22 22 27

Meaning it didn't throw/exit.

Maybe we could do a more simple check using something like:
```js
fs.existsSync(path.join(__dirname, 'package.json'))
```

What do you think?

**Comment by Tolsee** [comment]
@alex-saunders 
I think we can do something as you have recommended for now. But, when you said If there is a `package.json` in the parent directory that should mean that we are inside some application. And, I think in the long term it will be a good idea to let the user use glicky inside application even if they are not in the root of the directory.

**Comment by alex-saunders** [comment]
Good point @Tolsee , I hadn't thought about users wanting to run Glicky from not in the root directory, maybe we'll need to design a better system of checking whether the user is in a suitable directory for Glicky to be executed. However, I'm happy to merge this now, I'll just pull the branch down and check it's all working for me first.

**Comment by alex-saunders** [comment]
Screenshot 2019-04-20 at 09 18 12

Perfect! 🙌

**Pull request merged**